File: storage-mirror/android/src/main/java/at/bitfly/storagemirror/StorageMirrorPlugin.java
package at.bitfly.storagemirror;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "StorageMirror")
public class StorageMirrorPlugin extends Plugin {

    @PluginMethod
    public void echo(PluginCall call) {
        call.resolve(new JSObject());
    }
}

File: storage-mirror/ios/Sources/StorageMirrorPlugin/StorageMirrorPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(StorageMirrorPlugin)
public class StorageMirrorPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "StorageMirrorPlugin"
    public let jsName = "StorageMirror"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "reflect", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = StorageMirror()

    @objc func reflect(_ call: CAPPluginCall) {
        let keysToReflect = call.getArray("keys", String.self) ?? []
        implementation.reflect(keysToReflect)
        call.resolve()
    }
}
